// @framer/maikleenards.movie-posters/code/Movie_Poster.tsx
import * as React from "react";
import { Frame, PropertyControls, ControlType, Animatable } from "framer";

interface Props {
  width: number;
  height: number;
  searchType: boolean;
  apikey: string;
  search: string;
}

export class MoviePoster extends React.Component<Props> {
  static defaultProps = {
    url: null,
    apikey: null,
    search: "",
    searchType: "t",
    height: 290,
    width: 200,
  };

  static propertyControls: PropertyControls<Props> = {
    searchType: { type: ControlType.SegmentedEnum, title: "Search on", options: ["t", "i"], optionTitles: ["Name", "IMDb id"]},
    search: { type: ControlType.String, title: "Search", placeholder: "ex. Avengers" },
    apikey: { type: ControlType.String, title: "API Key", placeholder: "OMDb API Key"},
  };

  state = {
    url: null,
  };

//   componentDidMount() {
//       this.setImage({ search: 'Avengers' });
//   }

  componentWillReceiveProps(props: Props) {
    if (props.search !== this.props.search) this.setImage(props);
  }

  async setImage({ search, apikey, searchType }: Props) {
    const url = `http://www.omdbapi.com/?apikey=${apikey}&${searchType}=${search}`;
    fetch(url).then(data => data.json()).then(res => {
        this.setState({ url: res.Poster });
    });
  }

  render() {
    const { width, height } = this.props;

    return (
      <Frame width={width} height={height} background="white">
        <div
          style={{
            width: width,
            height: height,
            backgroundImage: `url(${this.state.url})`,
            backgroundSize: "cover",
            backgroundPosition: "center",
            display: "flex",
            justifyContent: "center",
            alignItems: "center",
          }}
        >
          <div
            style={{
              fontSize: "12px",
              color: "#999",
              textAlign: "center",
              zIndex: -1,
              position: "relative",
            }}
          >
            <p>Loading Poster</p>
          </div>
        </div>
      </Frame>
    );
  }
}
